x-pack/plugins/transform/public/app/sections/create_transform/components/group_by_list/popover_form.tsx

```diff
@@ -263,6 +280,31 @@ export const PopoverForm: React.FC<Props> = ({ defaultData, otherAggNames, onCha
           </>
         </EuiFormRow>
       )}
+      {isGroupByDateHistogram(defaultData) && timeZone ? (
+        <EuiFormRow
+          error={
+            !timeZoneValid && [
+              i18n.translate('xpack.transform.groupBy.popoverForm.timeZoneError', {
+                defaultMessage: 'Invalid time zone.',
+              }),
+            ]
+          }
+          isInvalid={!timeZoneValid}
+          label={i18n.translate('xpack.transform.groupBy.popoverForm.timeZoneLabel', {
+            defaultMessage: 'Time zone',
+          })}
+        >
+          <EuiSelect
```

Comment:
did you consider using `EuiCombobox` instead? there are quite some timezones, and combobox provides a better search experience.

Reply:
Updated here [`7aab47e` (#155535)](https://github.com/elastic/kibana/pull/155535/commits/7aab47ed136358cd9a479cb65c8e14f9c471e0fb)